overlord/snapstate/snapstate.go

```diff
@@ -901,6 +910,19 @@ func doUpdate(ctx context.Context, st *state.State, names []string, updates []*s
 		tasksets = append(tasksets, addAutoAliasesTs)
 	}
 
+	if len(rerefreshes) > 0 {
+		rerefresh := st.NewTask("rerefresh", fmt.Sprintf("Re-refresh %s due to epoch change", strutil.Quoted(rerefreshes)))
+		rerefresh.Set("rerefresh-setup", reRefreshSetup{
+			Snaps:  rerefreshes,
+			UserID: userID,
+			Flags:  globalFlags,
+		})
+		for _, taskset := range tasksets {
+			rerefresh.WaitAll(taskset)
```

Comment:
this is not the approach we discussed originally, we need to do this possibly even if some refreshes failed and some suscceeded, this will not do that